File: Php/app/Home/HomeModel.php
namespace app\Home{

	use app\Urls\HomegameAddUrlModel;
	use app\Admin\AdminNavModel;
	use core\HomegamePageModel;
	use entities\Cashgame;
	use entities\Homegame;
	use Domain\Classes\User;
	use app\Urls\UserAddUrlModel;
	use app\Urls\AuthLoginUrlModel;

	class HomeModel extends HomegamePageModel {

		public $isLoggedIn;
		public $addHomegameUrl;
		public $loginUrl;
		public $registerUrl;
		public $adminNav;

		public function __construct(User $user = null,
									Homegame $homegame = null,
									Cashgame $runningGame = null){
			parent::__construct($user, $homegame, $runningGame);
			isLoggedIn = $user != null;
			addHomegameUrl = new HomegameAddUrlModel();
			loginUrl = new AuthLoginUrlModel();
			registerUrl = new UserAddUrlModel();
			adminNav = new AdminNavModel($user);
		}

	}

}

File: Web/Models/HomeModel.cs
using Core.Classes;

namespace Web.Models{

	public class HomeModel : HomegamePageModel {

	    public bool IsLoggedIn { get; private set; }
	    public UrlModel AddHomegameUrl { get; private set; }
        public UrlModel LoginUrl { get; private set; }
        public UrlModel RegisterUrl { get; private set; }
	    public AdminNavModel AdminNav { get; private set; }

	    public HomeModel(User user, Homegame homegame, Cashgame runningGame) : base(user, homegame, runningGame)
	    {
			IsLoggedIn = user != null;
            AddHomegameUrl = new HomegameAddUrlModel();
            LoginUrl = new AuthLoginUrlModel();
            RegisterUrl = new UserAddUrlModel();
			AdminNav = new AdminNavModel(user);
	    }

	}

}